subsys/bluetooth/host/gatt.c

```diff
@@ -3828,14 +3827,7 @@ int bt_gatt_write_without_response_cb(struct bt_conn *conn, uint16_t handle,
 
 	cmd = net_buf_add(buf, sizeof(*cmd));
 	cmd->handle = sys_cpu_to_le16(handle);
-
-	write = net_buf_append_bytes(buf, length, data, K_NO_WAIT, NULL, NULL);
-	if (write != length) {
-		BT_WARN("Unable to allocate length %u: only %zu written",
-			length, write);
-		net_buf_unref(buf);
-		return -ENOMEM;
-	}
+	net_buf_add_mem(buf, data, length);
```

Comment:
Actually this assumption that the response cannot be bigger than then a single buffer is incorrect for EATT, EATT can in fact have channels which MTU is bigger then MPS which is why net_buf_append_bytes was used so it can allocate multiple buffers if necessary.

Reply:
@Vudentz As far as I can tell the buf allocated is sized according to the MTU, not the MPS, and when sending the ATT PDU then L2CAP will handle splitting the MTU up into MPS sized fragment if needed. To me this looks like something that is handled properly and does not include GATT/ATT.

The reason why I removed the net_buf_append_bytes was because allocate_cb argument was set to NULL, which means that if this call would have had to allocate more buffers it would have crashed, because the allocate_cb is not checked for non-NULL.

And by not having the return value of net_buf_append_bytes the write_buf callback could be simplified since it does not have to handle failures at the moment.

Reply:
@joerchan no it is not allocated according to the remote MTU, in EATT MTUs are not required to be symmetric, that would mean that one cannot take advantage of having bigger MTU because the code can only use a single buffer.

That said if net_buf_append_bytes does require a callback we should provide one, @asbjornsabo this might explain our crashes with EATT with some platforms as they might have bigger MTU then our buffer size.

Reply:
This should take care of the allocation callback being NULL: https://github.com/zephyrproject-rtos/zephyr/pull/31493